File: two-pointers/remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        slow = 0
        fast = 0

        while fast < len(nums):
            if nums[fast] != val:
                nums[slow] = nums[fast]
                slow+=1
            fast+=1
        return slow

File: array/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int slow = 0;
        int fast = 0;

        while (fast < nums.size()){
            if (nums[fast] != val){
                nums[slow++] = nums[fast];
            }
            fast++;
        }

        return slow;
    }
};